### PR TITLE
Tidy up logging module

### PR DESCRIPTION
**Pull request by milliams**
Primarily this is removing the `_get_logging` and `_get_handlers` functions which were used before to get access to the standard library `logging` module. This is replaceable with `absolute_import`.

Additionally it rearranges some imports, deletes unused code and removes unused arguments from some functions.

I realise we are moving towards deprecating this module but in the meantime it's worth reducing the amount of cruft.

### CODE REVIEW

File: python/Ganga/Utility/logging/__init__.py [hunk 1]
```diff
@@ -80,7 +59,7 @@ def _get_handlers():
 private_logger = None  # private logger of this module
 
 # main logger corresponds to the root of the hierarchy
-main_logger = _get_logging().getLogger()
+main_logger = logging.getLogger()
```
Comment (rob-c) [inline python/Ganga/Utility/logging/__init__.py]: Are we certain that the ```main_logger``` is never multiply defined when the class is imported multiple times?

I had a lot of grief getting the logger to shut down correctly as the final object in the Shutdown manager a while ago (hence a lot of the changes for ```_get_logging``` and I'd like to make sure it gracefully shuts down now)

**Comment by rob-c** [comment]
I'd be tempted to remove https://github.com/ganga-devs/ganga/pull/334/files#diff-cafdab496baeebb8d12db8dfdd0c8e8eR313 While we're here making changes. I often end up having to do this to work out where an actual error comes from.
(I run my logging formatted with line numbers and the 'filename' as parsed by ganga)

File: python/Ganga/Utility/logging/__init__.py (hunk 1 above)
Comment (milliams) [inline python/Ganga/Utility/logging/__init__.py]: A module will not be imported multiple times as it is cached in `sys.modules`. Unless we have a very messed up `PYTHONPATH` we should be safe.

**Comment by milliams** [comment]
Yes, I've often had to do the same to be able to track down the source of errors but we need to ensure that:

1. It doesn't make the printout too verbose for users
2. it doesn't break anything where users have set custom logging levels for various modules.

I agree that in general `_guess_module_logger_name` could use some simplification but I probably don't have a good enough understanding of it to do so just yet.

**Comment by rob-c** [comment]
I'll add a config option to control that line if we don't want to inflict it upon users, but I strongly think it would help speed up understanding what has gone on when users post their terminal capture asking for help.

**Comment by rob-c** [comment]
I'd prefer to see the 'remove filename' from the logger removed unless there are strong objections?
Other than this I'd say go for merge :+1:

**Comment by milliams** [comment]
I don't have an objection in principle but I'll leave it to another PR to avoid holding this one up. It might cause some discussion so I'd rather keep that separate.

**Comment by rob-c** [comment]
In the interest of moving forward OK, I'll open another PR for that

**Pull request merged**